src/kv/lazy-store.ts

```diff
@@ -0,0 +1,202 @@
+import {RWLock} from '../rw-lock';
+import type {Read, Store, Value, Write} from './store';
+import {deleteSentinel, WriteImplBase} from './write-impl-base';
+import {getSizeOfValue} from '../get-size-of-value';
+
+export class LazyStore implements Store {
+  private readonly _cache: Cache;
+  private readonly _rwLock = new RWLock();
+  private readonly _base: Store;
+
+  constructor(
+    base: Store,
+    cacheSizeLimit: number,
+    shouldBePinned: (key: string) => boolean,
+  ) {
+    this._base = base;
+    this._cache = new Cache(cacheSizeLimit, shouldBePinned);
+  }
+
+  async read(): Promise<Read> {
+    const release = await this._rwLock.read();
+    return new ReadImpl(this._cache, this._base, release);
+  }
+
+  async withRead<R>(fn: (read: Read) => R | Promise<R>): Promise<R> {
+    const read = await this.read();
+    try {
+      return await fn(read);
+    } finally {
+      read.release();
+    }
+  }
+
+  async write(): Promise<Write> {
+    const release = await this._rwLock.write();
+    return new WriteImpl(
+      this._cache,
+      new ReadImpl(this._cache, this._base, release),
+    );
+  }
+
+  async withWrite<R>(fn: (write: Write) => R | Promise<R>): Promise<R> {
+    const write = await this.write();
+    try {
+      return await fn(write);
+    } finally {
+      write.release();
+    }
+  }
+
+  close(): Promise<void> {
+    return this._base.close();
+  }
+
+  get closed(): boolean {
+    return this._base.closed;
+  }
+}
+
+class ReadImpl implements Read {
+  private readonly _cache: Cache;
+  private readonly _release: () => void;
+  private readonly _base: Store;
+  private _baseRead: Read | undefined;
+  private _closed = false;
+
+  constructor(cache: Cache, base: Store, release: () => void) {
+    this._cache = cache;
+    this._base = base;
+    this._release = release;
+  }
+
+  release() {
+    if (this._baseRead) {
+      this._baseRead.release();
+    }
+    this._release();
+    this._closed = true;
+  }
+
+  get closed(): boolean {
+    return this._closed;
+  }
+
+  async has(key: string): Promise<boolean> {
+    return this._cache.has(key) || (await this.getBaseRead()).has(key);
+  }
+
+  async get(key: string): Promise<Value | undefined> {
+    let v = this._cache.get(key);
+    if (v === undefined) {
+      v = await (await this.getBaseRead()).get(key);
+      if (v !== undefined) {
+        this._cache.set(key, v);
+      }
+    }
+    return v;
+  }
+
+  private async getBaseRead(): Promise<Read> {
```

Comment:
leading underscore for private methods.

Can you fix the eslint config to catch this?